alf/algorithms/agent.py

```diff
@@ -233,7 +233,9 @@ def train_step(self, exp: Experience, state):
             new_state = new_state._replace(icm=icm_step.state)
 
         rl_step = self._rl_algorithm.train_step(
-            exp._replace(observation=observation), state.rl)
+            exp._replace(
+                observation=observation, rollout_info=exp.rollout_info.rl),
```

Comment:
Why do we still need to put rollout_info in train_step if now summarize_rollout is separated from summarize_train? From my understanding, if we want to summarize rollout statistics, we can only do it once when collecting experiences.

Reply:
Some algorithms need rollout_info (e.g. PPO)

Reply:
I mean whether updating `rollout_info=exp.rollout_info.rl` here won't matter as this field will no longer be used later in the code (not for summary anymore), right?

Reply:
I mean this update `rollout_info=exp.rollout_info.rl` here is meaningless because it won't be used later in `OnPolicyAlgorithm train_step`

Reply:
An algorithm can choose what ever it does with rollout_info. There is no guarantee it will not be used.